internal/policies/apparmor/apparmor.go

```diff
@@ -0,0 +1,262 @@
+		o(&args)
+	}
+
+	return &Manager{
+		pathsMu:           make(map[string]*sync.Mutex),
+		apparmorDir:       apparmorDir,
+		apparmorParserCmd: args.apparmorParserCmd,
+	}, nil
+}
+
+// AssetsDumper is a function which uncompress policies assets to a directory.
+type AssetsDumper func(ctx context.Context, relSrc, dest string, uid int, gid int) (err error)
+
+// ApplyPolicy generates an apparmor policy based on a list of entries.
+// Steps:
+// 1. Move /etc/apparmor.d/adsys/<object> to /etc/apparmor.d/adsys/<object>.old
+// 2. Create /etc/apparmor.d/adsys/<object> with new policy
+// 3. Run apparmor_parser -r on all files in /etc/apparmor.d/adsys/<object>
+// 4a. If apparmor_parser fails, move /etc/apparmor.d/adsys/<object>.old to /etc/apparmor.d/adsys/<object>
+// 4b. If apparmor_parser succeeds, remove /etc/apparmor.d/adsys/<object>.old.
+func (m *Manager) ApplyPolicy(ctx context.Context, objectName string, isComputer bool, entries []entry.Entry, assetsDumper AssetsDumper) (err error) {
+	defer decorate.OnError(&err, i18n.G("can't apply apparmor policy to %s"), objectName)
+
+	// TODO: add user support
+	// Apparmor policies are currently only supported on computers
+	if !isComputer {
+		return nil
+	}
+
+	log.Debugf(ctx, "Applying apparmor policy to %s", objectName)
+
+	objectDir := "machine"
+	if !isComputer {
+		// TODO: add user support
+		objectDir = "users"
+	}
+	apparmorPath := filepath.Join(m.apparmorDir, objectDir)
+
+	// Mutex is per destination path (e.g. machine, users/user1, users/user2)
+	m.muMu.Lock()
+	// if mutex does not exist for this destination, creates it
+	if _, exists := m.pathsMu[apparmorPath]; !exists {
+		m.pathsMu[apparmorPath] = &sync.Mutex{}
+	}
+	m.muMu.Unlock()
+	m.pathsMu[apparmorPath].Lock()
+	defer m.pathsMu[apparmorPath].Unlock()
+
+	// If we have no entries, attempt to unload them and remove the apparmor directory
+	if len(entries) == 0 {
+		return unloadRules(ctx, m.apparmorParserCmd, apparmorPath)
+	}
+
+	if err := os.MkdirAll(apparmorPath, 0750); err != nil {
+		return fmt.Errorf(i18n.G("can't create apparmor directory %q: %v"), apparmorPath, err)
+	}
+
+	// Remove any existing .old directory
+	oldApparmorPath := apparmorPath + ".old"
+	if err := os.RemoveAll(oldApparmorPath); err != nil {
+		return fmt.Errorf(i18n.G("can't remove old apparmor directory %q: %v"), oldApparmorPath, err)
+	}
+	// Rename existing apparmor policy to .old
+	if err := os.Rename(apparmorPath, oldApparmorPath); err != nil {
+		return fmt.Errorf(i18n.G("can't rename apparmor directory %q to %q: %v"), apparmorPath, oldApparmorPath, err)
+	}
+	defer cleanupApparmorDir(ctx, oldApparmorPath, apparmorPath)
+
+	// Dump assets to the adsys/machine/ subdirectory with correct ownership. If no assets is present while entries != nil, we want to return an error.
+	if err := assetsDumper(ctx, "scripts/apparmor/", apparmorPath, -1, -1); err != nil {
+		return err
+	}
+
+	// Get the list of files to run apparmor_parser on
+	e := entries[0]
```

Comment:
I understand why you take the first entry, but I would maybe safeguard this on the key name. Imagine that someone messed with the admx/adml and then, the pol files creates 2 keys, we want to act at least on the well-known name.